src/Redux/book/book.js

```diff
@@ -1,22 +1,70 @@
-import bookList from '../../fakeBooksData';
-// Actions
+import booksURL from '../apiURL';
+
+// ACTIONS
+const GET_BOOKLIST = 'GET_BOOKLIST';
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
 
-// Action Creators
-
-export const addBook = (payload) => ({ type: ADD_BOOK, payload });
-export const removeBook = (payload) => ({ type: REMOVE_BOOK, payload });
-
-// Reducer
-const initialState = bookList;
+// ACTION CREATORS ------------------------------------------
+// Get book list
+export const getBooklist = () => async (dispatch) => {
+  const response = await fetch(booksURL);
+  const data = await response.json();
+  const bookList = [];
+  Object.keys(data).forEach((ID) => {
+    bookList.push({ ...data[ID][0], item_id: ID });
+  });
+  if (response.ok) {
+    dispatch({
+      type: GET_BOOKLIST,
+      payload: bookList,
+    });
+  }
+};
+// Add book
+export const addBook = (payload) => async (dispatch) => {
+  const response = await fetch(booksURL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: payload.item_id,
+      title: payload.title,
+      author: payload.author,
+      category: payload.category,
+    }),
+  });
+  if (response.ok) {
+    dispatch({ type: ADD_BOOK, payload });
+  }
+};
+// Remove book
+export const removeBook = (payload) => async (dispatch) => {
+  const response = await fetch(`${booksURL}/${payload}`, {
+    method: 'DELETE',
```

Comment:
Hi @BenMukebo!

First of all, thanks for taking the time to review the code. I really appreciate it.

About this issue, I believe that in the requirements they give us the option to work with fetch OR axios with AsyncThunk, as seen in the [link](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.)

A copy from the page here:
"Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls."

After I get approved and If I have time I'll use this to learn it better, but my approval musn't depend on it, since I had another option.

I hope you undersand it.